### ISSUE TITLE
Can you explain the loss function (5) in paper?

### ISSUE THREAD

**Issue by csldali** [jwchoi384/Gaussian_YOLOv3#11]
Hi, nice work!

I want to use tf implementation the work, but i don`t know loss function (5) in paper; 
Especially the vertical line symbol in Especially the vertical line symbol；
I read you code, but i don`t know the following code:

`    float temp_x = (1./2.) * 1./(normal_dist_x+epsi) * normal_dist_x * scale;
    float temp_y = (1./2.) * 1./(normal_dist_y+epsi) * normal_dist_y * scale;
    float temp_w = (1./2.) * 1./(normal_dist_w+epsi) * normal_dist_w * scale;
    float temp_h = (1./2.) * 1./(normal_dist_h+epsi) * normal_dist_h * scale;

    delta[index + 0*stride] = temp_x * in_exp_x  * (1./x[index+1*stride]);
    delta[index + 2*stride] = temp_y * in_exp_y  * (1./x[index+3*stride]);
    delta[index + 4*stride] = temp_w * in_exp_w  * (1./x[index+5*stride]);
    delta[index + 6*stride] = temp_h * in_exp_h  * (1./x[index+7*stride]);

    delta[index + 1*stride] = temp_x * (in_exp_x_2/x[index+1*stride] - 1./(x[index+1*stride]+sigma_const));
    delta[index + 3*stride] = temp_y * (in_exp_y_2/x[index+3*stride] - 1./(x[index+3*stride]+sigma_const));
    delta[index + 5*stride] = temp_w * (in_exp_w_2/x[index+5*stride] - 1./(x[index+5*stride]+sigma_const));
    delta[index + 7*stride] = temp_h * (in_exp_h_2/x[index+7*stride] - 1./(x[index+7*stride]+sigma_const));`

Can you give a detailed explanation? Thanks for your great work.

**Comment by xiaowenhe** [comment]
Also confused about the loss function, can give a detailed explanation? Thank you!
@jwchoi384

**Comment by jwchoi384** [comment]
@csldali @xiaowenhe 
Please check this: [The probability density function for the normal distribution](https://medium.com/@karthik.k183/normal-distribution-fd601166db0b)

`delta[index + ~]` means gradient.
So, we need to calculate the gradient for mean and variance of NLL loss function.
In the case of TF, the back propagation is easily implemented, but in C implementation, we calculate the gradient and do coding.
 
If you calculate the gradient for mean and variance of the probability density function for the normal distribution, you can understand what I mean.
(For stable training, I added the `sigma_const`, which is a hyper parameter.)
